sim/core/attack.go

```diff
@@ -653,15 +744,57 @@ func (aa *AutoAttacks) UpdateSwingTimers(sim *Simulation) {
 	}
 }
 
+func (aa *AutoAttacks) ExtraMHAttackProc(sim *Simulation, attacks int32, actionID ActionID, spell *Spell) {
+	if spell.Flags.Matches(SpellFlagBatchStopAttackMacro) {
+		aa.StoreExtraMHAttack(sim, attacks, actionID, spell.ActionID)
+	} else {
+		aa.ExtraMHAttack(sim, attacks, actionID, spell.ActionID)
+	}
+}
+
 // ExtraMHAttack should be used for all "extra attack" procs in Classic Era versions, including Wild Strikes and Hand of Justice. In vanilla, these procs don't actually grant a full extra attack, but instead just advance the MH swing timer.
-func (aa *AutoAttacks) ExtraMHAttack(sim *Simulation, attacks int32, actionID ActionID) {
+func (aa *AutoAttacks) ExtraMHAttack(sim *Simulation, attacks int32, actionID ActionID, triggerAction ActionID) {
+    if attacks == 0 {
+		return
+    }
 	if sim.Log != nil {
-		aa.mh.unit.Log(sim, "gains %d extra attacks from %s", attacks, actionID)
+		aa.mh.unit.Log(sim, "gains %d extra attacks from %s triggered by %s", attacks, actionID, triggerAction)
 	}
 	aa.mh.swingAt = sim.CurrentTime + SpellBatchWindow
 	aa.mh.spell.SetMetricsSplit(1)
 	sim.rescheduleWeaponAttack(aa.mh.swingAt)
-	aa.mh.extraAttacks += attacks
+	aa.mh.extraAttacksPending += attacks
+}
+
+func (aa *AutoAttacks) StoreExtraMHAttack(sim *Simulation, attacks int32, actionID ActionID, triggerAction ActionID) {
+    if attacks == 0 {
+        return
+    }
+
+	if aa.mh.extraAttacksAura == nil {
+		aa.mh.extraAttacksAura = aa.mh.unit.GetOrRegisterAura(Aura{
+			Label:     "Extra Attacks", // Tracks Stored Extra Attacks from all sources
+			ActionID:   ActionID{SpellID: 21919}, // Thrash ID
+			Duration:  NeverExpires,
+			MaxStacks: 4, // Max is 4 extra attacks stored - more can proc after
+			OnGain: func(aura *Aura, sim *Simulation) {
+			},
+			OnExpire: func(aura *Aura, sim *Simulation) {
+			},
```

Comment:
These callbacks should be able to be removed if they're unused

Reply:
removing